civismlext/stacking.py

```diff
@@ -1,15 +1,37 @@
 from __future__ import print_function
 from __future__ import division
 
-import numpy as np
 from abc import ABCMeta, abstractmethod
+import warnings
+
+import numpy as np
 import six
 from sklearn.base import BaseEstimator, clone
 from sklearn.utils.metaestimators import if_delegate_has_method
 from sklearn.model_selection import check_cv
 from sklearn.utils import tosequence, check_X_y
 from sklearn.externals.joblib import Parallel, delayed
-from sklearn.model_selection._validation import _index_param_value
+
+try:
+    # TODO: Avoid using a private function from scikit-learn.
+    #  _check_fit_params was added at sklearn 0.22.1
+    from sklearn.utils.validation import _check_fit_params
+except ImportError:
+    # _index_param_value was removed in sklearn 0.22.1
+    # See: https://github.com/scikit-learn/scikit-learn/pull/15863
+    from sklearn.model_selection._validation import _index_param_value
+
+    warnings.warn(
```

Comment:
Quick q: this is best practice for emitting warnings vs. using the `logging` package?

Reply:
No strong preference from me, though we've started to follow the Python data science stack for using `FutureWarning` via `warnings.warn`, e.g., https://github.com/civisanalytics/civis-python/pull/340.

Reply:
Yeah, after looking through, it seems we use this elsewhere so it's fine by me